File: DTOs/Sources/NotificationsServiceDTOs/RegisterNotificationDeviceDTO.swift
public struct RegisterNotificationDeviceDTO: Codable {
    public var deviceID: String
    public var token: String
    public var platform: NotificationPlatform

    public init(
        deviceID: String,
        token: String,
        platform: NotificationPlatform
    ) {
        self.deviceID = deviceID
        self.token = token
        self.platform = platform
    }
}

public enum NotificationPlatform: String, Codable, CaseIterable {
    case android
    case ios
}

File: android/app/src/main/java/net/ipv64/kivop/dtos/NotificationsServiceDTOs/RegisterNotificationDeviceDTO.kt
package net.ipv64.kivop.dtos.NotificationsServiceDTOs

import java.util.UUID
import java.time.LocalDateTime

data class RegisterNotificationDeviceDTO (
    var deviceID : String,
    var token : String,
    var platform : NotificationPlatform,
)
enum class NotificationPlatform {
    android,
    ios,
}
